src/components/CardModal/CardModalActions/MoveMenu/MoveMenu.tsx

```diff
@@ -19,14 +20,20 @@ const MoveMenu = () => {
     if (!card) return;
 
     dispatch(moveCardToList({ cardId: card.id, destListId: selectedListId }));
+    dispatch(setCurrentSelectedCard({ ...card, listId: selectedListId }));
```

Comment:
This is quite important and something I missed. `currentSelectedCard` controls the card data associated with the card modal. If this is out of sync at any point, updates may not be reflected correctly. 

Previously it updated the cards state to update the list id of the card. However, this doesn't automatically update `currentSelectedCard`. To do that, you need to set it